src/recorded_battle.c

```diff
@@ -87,7 +87,7 @@ void RecordedBattle_Init(u8 mode)
             for (j = 0; j < BATTLER_RECORD_SIZE; j++)
                 sBattleRecords[i][j] = 0xFF;
             sBattleFlags = gBattleTypeFlags;
-            sAI_Scripts = gBattleResources->ai->aiFlags;
+            sAI_Scripts = gBattleResources->ai->aiFlags[B_POSITION_OPPONENT_LEFT];
```

Comment:
Hmm, I worry about how this will affect recorded battles, as I'm unfamiliar with how AI is handled for those, if at all.

Reply:
Any pre-existing recorded battles controlled by AI where the battlers now would have different AI scripts is reasonably likely to crash. I'm not sure if people are likely to have any recorded battles like that tbh.

In any case, it probably makes sense to extend `sAI_Scripts` to have an entry for each battler, and extend `GetAiScriptsInRecordedBattle` to take a `battlerId`. But that also breaks existing recorded battles unless we take care to ensure the saves remain binary compatible.

Extending recorded battles to support AI flags per battler would also mean the AI tests could use that (although I can't think of any practical reason to have different flags per battler).

Reply:
So I'm not 100% sure on if this can happen in-game, but if you have a recorded battle of something that's normally controlled by the AI, then iirc the game just runs the AI rather than recording all the choices. (I noticed this behavior when I was first building the test system, but haven't dug into it—this is basically the reason why we can't test wild Pokémon, they're controlled by the AI so we can't make them do what we want [although these days with a little RNG work we could])

wrt practically, I just meant within tests. I agree with your in-game reasons for wanting it :)